src/Ng/CommandHelpers.cs

```diff
@@ -127,7 +127,10 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             return CreateStorageFactoryImpl(arguments, names, verbose);
         }
 
-        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, IDictionary<string, string> argumentNameMap, bool verbose)
+        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, 
+                                                               IDictionary<string, string> argumentNameMap,
+                                                               bool verbose,
+                                                               bool compressed = false)
```

Comment:
There are different types of parameters. The ones from dictionary are mapped form external arguments. This is an internal argument. I let it at it is for now.